### ISSUE TITLE
installation of coqide 8.10 via opam fails due to missing handling of camlp5dir

### ISSUE THREAD

**Issue by eladrion** [coq/coq#9916]
Dear Coq developers,
I just wanted to install the current Coq and CoqIDE development version from the official repo via opam. Coq itself is installed successfully but the installation of CoqIDE fails.

You can find the installation process trace at the bottom.

What I found out so far is that the opam definition file for CoqIDE defines a dependency to camlp5 and setting of the camlp5 library directory accordingly. But the respective configure.ml does not handle the argument "camlp5dir".

Regards,
Mario Frank

<><> Gathering sources ><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[coqide.dev] synchronised from git+https://github.com/coq/coq.git#master

<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[ERROR] The compilation of coqide failed at "/home/eladrion/.opam_coq_810/opam-init/hooks/sandbox.sh build ./configure -configdir /home/eladrion/.opam_coq_810/default/lib/coq/config -prefix
        /home/eladrion/.opam_coq_810/default -mandir /home/eladrion/.opam_coq_810/default/man -docdir /home/eladrion/.opam_coq_810/default/doc -libdir /home/eladrion/.opam_coq_810/default/lib/coq -datadir
        /home/eladrion/.opam_coq_810/default/share/coq -camlp5dir /home/eladrion/.opam_coq_810/default/lib/camlp5".

#=== ERROR while compiling coqide.dev =========================================#
# context     2.0.3 | linux/x86_64 | ocaml-system.4.07.1 | http://coq.inria.fr/opam/core-dev#2019-04-04 14:20
# path        ~/.opam_coq_810/default/.opam-switch/build/coqide.dev
# command     ~/.opam_coq_810/opam-init/hooks/sandbox.sh build ./configure -configdir /home/eladrion/.opam_coq_810/default/lib/coq/config -prefix /home/eladrion/.opam_coq_810/default -mandir /home/eladrion/.opam_coq_810/default/man -docdir /home/eladrion/.opam_coq_810/default/doc -libdir /home/eladrion/.opam_coq_810/default/lib/coq -datadir /home/eladrion/.opam_coq_810/default/share/coq -camlp5dir /home/eladrion/.opam_coq_810/default/lib/camlp5
# exit-code   2
# env-file    ~/.opam_coq_810/log/coqide-13302-6cc03f.env
# output-file ~/.opam_coq_810/log/coqide-13302-6cc03f.out
### output ###
# Available options for configure are:
# [...]
#   -bytecode-compiler (yes|no) Enable Coq's bytecode reduction machine (VM)
#   -native-compiler (yes|no)   Compilation to native code for conversion and normalization
#   -coqwebsite                 URL of the coq website
#   -force-caml-version         Force OCaml version
#   -force-findlib-version      Force findlib version
#   -warn-error (yes|no)        Make OCaml warnings into errors (default no)
#   -camldir <dir>              Specifies path to 'ocaml' for running configure script
#   -profile <profile>          Sets a bunch of flags. Supported profiles:
#      devel = -local -annot -bin-annot -warn-error yes
#   -help                       Display this list of options
#   --help                      Display this list of options

**Comment by ejgallego** [comment]
Umm, does `opam pin add coqide git+https://github.com/coq/coq.git` help?

**Comment by ejgallego** [comment]
But indeed that's a bug in opam repository I think [either in the main one or in the one in Coq]

Can you post the output of `opam repos`?

**Comment by eladrion** [comment]
No, pinning does not help. I get a new error, saying:
´The following dependencies couldn't be met:
  - coqide → coqide-server
      unknown package
´

The repos are:
- coq-core-dev http://coq.inria.fr/opam/core-dev
- default      https://opam.ocaml.org

**Comment by ejgallego** [comment]
Yup that seems like a bug in https://github.com/coq/opam-coq-archive/blob/master/core-dev/packages/coqide/coqide.dev/opam I suggest we just submit a PR removing the offending line.

> No, pinning does not help.

Ups, that's unfortunate that you cannot pin a package it doesn't know about [at least it couldn't not so long ago]
